src/test/spec/unified_runner/entity.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{bson::Bson, test::util::EventClient, Collection, Database};
+
+pub enum Entity {
+    Client(EventClient),
+    Database(Database),
+    Collection(Collection),
+    Result(Option<Bson>),
```

Comment:
Is there a reason to have `None` be a Result instead of a separate variant, e.g. `Result(Bson)` and `None`?

Reply:
Hm, I hadn't thought about doing that. The purpose of this variant is to store return values from operations as entities so that they can be used later on, so it makes the most sense to me to have one variant to represent that instead making a differentiation between an operation that returns a Bson value and an operation that returns `None`. Since we don't have change streams in the driver, though, it's hard to know what the most convenient representation would be for actually using these values since they're mainly intended for change stream tests.

Reply:
I might just not be understanding the context for this variant correctly, but I'd think that we'd generally want to execute different code paths when we have `Some(bson_val)` and None. If it's None, I imagine if it's `Some(bson_val)`, we'd presumably be doing something with it, and if it's None, we wouldn't be doing much with it at all. To put it another way, any time we have an Option<Bson>, is the first thing we're going to do with it be checking if it's Some or None and then taking a separate code path? If so, then there might not be much benefit to having them be the same variant.

Reply:
[Q&A on the spec PR](https://github.com/mongodb/specifications/pull/846#discussion_r486568386), a re-read of the `saveResultAsEntity` section made me question when we'd use this functionality in the first place.

From the discussion here, it appears that storing these results is not useful in the current iteration of the spec. I see what you're saying, and if there were some operation introduced in the future it would probably be along the lines of what you described, so I'm open to splitting this out into two variants and can revisit the the design if needed if there is an option introduced that uses them in the future.